### ISSUE TITLE
RTP support for the output of the sound card 

### ISSUE THREAD

**Issue by vojenfr** [duncanthrax/scream#11]
Hello @duncanthrax,
It's possible to add a RTP/DLNA support for the output instead of raw PCM stream?
So great driver!
Thanks you

**Comment by duncanthrax** [comment]
Hi @vojenfr,
these are separate things:

* Adding RTP support should be easy. RTP supports 2-channel linear PCM as a default format, so it's probably just a matter of prepending the correct header to the data.

* I'll not add DLNA advertisement support, since Scream is a pure kernel solution, and DLNA is too complex to implement in a kernel driver.

Adding RTP would have the advantage that the stream can be read by standard software, e.g. VLC. Is that what you want? What is your use case?

/tom

**Comment by vojenfr** [comment]
Yes exactly @duncanthrax , I would like to listen with `vlc rtp://239.255.77.77:4010` .
And also redirect directly on a amplifier :blush:
It is scheduled in the next release?

**Comment by duncanthrax** [comment]
I have it kind of working. There's some additional overhead because RTP expects PCM in network byte order.

Also VLC does not like picking up a restarting stream. I might have to send silence instead of stopping the stream when no audio is playing.

Finally, it seems that VLC adds quite some buffering and therefore delay. That might be fixable with config parameters.

If I go through with this, I'll offer an extra driver with RTP support.
